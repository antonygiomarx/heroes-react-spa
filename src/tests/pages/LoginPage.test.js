const { mount } = require("enzyme");
const { AuthContext } = require("../../auth/authContext");
const { LoginPage } = require("../../pages/Login/LoginPage");
const { types } = require("../../types/types");

describe('Test in Login Page', () => {
    
    const history = {
        push: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn()
    }

    const wrapper = mount(
        <AuthContext.Provider value={ contextValue }>
            <LoginPage history={history} />
        </AuthContext.Provider>
        
    )

    test('should show itself correctly', () => {
        
        expect( wrapper ).toMatchSnapshot()

    });

    test('should to do the dispatch and navigation', () => {
        
        wrapper.find('button').simulate('click')

        expect( contextValue.dispatch ).toHaveBeenCalledWith({
            type: types.login,
            payload: { name: 'Erik A' }
        });
        expect( history.push ).toHaveBeenCalledWith('/')

        localStorage.setItem('lastPath', '/dc')
        wrapper.find('button').simulate('click')
        expect( history.push ).toHaveBeenCalledWith('/dc')
    });

});