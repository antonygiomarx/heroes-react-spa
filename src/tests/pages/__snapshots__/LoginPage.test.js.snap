// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Login Page should show itself correctly 1`] = `
<div
  className="container"
>
  <main>
    <header>
      <h1>
         Log in 
      </h1>
    </header>
    <hr />
    <button
      className="btn-logout"
      onClick={[Function]}
    >
      Log In
    </button>
  </main>
</div>
`;
